### DISCUSSION TITLE
The calculation problem of three bands of BVR.

### DISCUSSION THREAD

**Discussion by wpr1820** [phoebe-project/phoebe2#727] [Questions]
Hello, Sir.

I have a problem. When I input the three BVR bands together into PHOEBE2.4, the V band fits very well. But the zero phase in the B and R bands is not very good. We changed the comparison star and the check star and got the same result. 

When entering, we described the BVR data:
_b = phoebe.default_binary(contact_binary=True)
b.add_dataset('lc', times=T1,fluxes=F1,sigmas=S1,dataset='lcB',passband='Johnson:B',compute_phases=phoebe.linspace(0,1,101))
b.add_dataset('lc', times=T2,fluxes=F2,sigmas=S2,dataset='lcV',passband='Johnson:V',compute_phases=phoebe.linspace(0,1,101))
b.add_dataset('lc', times=T3,fluxes=F3,sigmas=S3,dataset='lcR',passband='Johnson:R',compute_phases=phoebe.linspace(0,1,101))
b.set_value_all('ld_mode@lcB', 'lookup')
b.set_value_all('ld_mode@lcV', 'lookup')
b.set_value_all('ld_mode@lcR', 'lookup')_

Do you know what the reason is? Thank you very much.
Here are our results

![B](https://user-images.githubusercontent.com/112557810/232993994-84e1015d-3eb8-4d3c-9b91-756c3e579dca.png)
![R](https://user-images.githubusercontent.com/112557810/232994007-0dcfc613-04ab-4944-a950-f65099ed66de.png)
![V](https://user-images.githubusercontent.com/112557810/232994013-f9968aa0-ab87-4bb5-adde-64f5de99e8bc.png)

**Comment by kecnry** [comment]
What are you using for [pblum_mode](http://phoebe-project.org/docs/latest/tutorials/pblum) and [l3 (third light)](http://phoebe-project.org/docs/latest/tutorials/l3)?  If your data are absolutely calibrated, then you might want to consider setting the `pblum_mode` for the first to `component-coupled` and the remaining to `dataset-coupled` so they follow the appropriate scalings relative to each other.  Otherwise you'll need to set the appropriate values for each manually to get the scaling correct.

Are you attempting to fit these (with an optimizer or sampler) or just running the forward model with known inputs?

**Comment by wpr1820** [comment]
That's what the optimizer calculated, we didn't shift any data. Does PHOEBE's input data have to be absolute flux？



---Original---
From: "Kyle ***@***.***&gt;
Date: Wed, Apr 19, 2023 21:57 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [phoebe-project/phoebe2] The calculation problem of three bandsof BVR. (Discussion #727)




 
What are you using for pblum_mode and l3 (third light)?  If your data are absolutely calibrated, then you might want to consider setting the pblum_mode for the first to component-coupled and the remaining to dataset-coupled so they follow the appropriate scalings relative to each other.  Otherwise you'll need to set the appropriate values for each manually to get the scaling correct.
 
Are you attempting to fit these (with an optimizer or sampler) or just running the forward model with known inputs?

**Reply by kecnry** [reply]
If you want to make use of coupled luminosities (see the links I sent above), then your data needs to be appropriately scaled relative to each other.  Otherwise, if your input observations are scaled, then you'll just need to recognize that any fitted values for the luminosity are also scaled and not in absolute units.  So if appropriate, I'd suggest making use of the dataset-coupling and then optimizing for a single luminosity.  Alternatively you can fit each of the luminosities separately if you are willing to lose the "color" information.
